server/server.go

```diff
@@ -80,43 +84,100 @@ func (s *Server) Shutdown(ctx context.Context) error {
 }
 
 // Handlers will return handlers
-func (s *Server) Handlers() *nrhttprouter.Router {
+func (s *Server) Handlers() *gin.Engine {
 	// Start a transaction for loading handlers
 	txn := s.Services.NewRelic.StartTransaction("load_handlers")
 	defer txn.End()
 
-	// Create a new router
 	segment := txn.StartSegment("create_router")
-	s.Router = apirouter.NewWithNewRelic(s.Services.NewRelic)
-	s.Router.HTTPRouter.Handler(http.MethodGet, "/swagger", http.RedirectHandler("/swagger/index.html", http.StatusMovedPermanently))
-	s.Router.HTTPRouter.Handler(http.MethodGet, "/swagger/*any", httpSwagger.WrapHandler)
-	if metrics, enabled := metrics.Get(); enabled {
-		s.Router.HTTPRouter.Handler(http.MethodGet, "/metrics", metrics.HTTPHandler())
+
+	httpLogger := s.Services.Logger.With().Str("service", "http-server").Logger()
+	engine := gin.New()
+	engine.Use(gin.LoggerWithWriter(debugWriter(&httpLogger)), gin.Recovery())
+	engine.Use(auth.CorsMiddleware())
+
+	if httpLogger.GetLevel() > zerolog.DebugLevel {
+		gin.SetMode(gin.ReleaseMode)
 	}
-	segment.End()
+	logging.SetGinWriters(&httpLogger)
+	engine.Use(logging.GinMiddleware(&httpLogger), gin.Recovery())
```

Comment:
Please change it to:
```golang
if httpLogger.GetLevel() > zerolog.DebugLevel {
    gin.SetMode(gin.ReleaseMode)
}
logging.SetGinWriters(&httpLogger)
engine := gin.New()
engine.Use(logging.GinMiddleware(&httpLogger), gin.Recovery())
engine.Use(auth.CorsMiddleware())
```

- `logging.SetGinWriters` should be before `gin.New` because is sets global `gin` variables. 
- `gin.SetMode(gin.ReleaseMode)` also should be before `gin.New`
- Setting logging&recovery middleware is currently done twice (which is wrong) - Just put the new `engine.Use(logging.GinMiddleware(&httpLogger), gin.Recovery())` right after `gin.New` and remove `engine.Use(gin.LoggerWithWriter(debugWriter(&httpLogger)), gin.Recovery())`

Reply:
Done